### DISCUSSION TITLE
How can I change the layout of the html page (table) to a gallery mode without processing the queries to the server ?

### DISCUSSION THREAD

**Discussion by joffreypersia** [livewire/livewire#6796] [1. Help] [answered]
Hi, 

I am building a web app that is a tool for photographers. The server has a lot of images and I need to be cautious with the queries I send. 

I have a page that displays a table, and I would like to allow the user to change the view to a gallery view. It will be the same data, filtered or not, the same pagination, just a different view. 

What might be the best strategy? 

At first, I wanted to use 2 different Livewire view components (one for table view and one for gallery view) with the same livewire class that will return $items to each view. 
But after searching inside the doc, there is a new render, so there is a new lifecycle and a new request to the query. 

So, I feel there are only two options available : 
- Render everything in the DOM and with Alpine, play with the [x-show](https://alpinejs.dev/directives/show). The only downside I see is that there will be a heavier DOM
- Use the [Laravel cache feature](https://livewire.laravel.com/docs/computed-properties#caching-between-requests) to store all the $items and then use Livewire to display the table/gallery conditionally. 

What might be your approach and why? 
Thank you so much
PS: I am using the same structure for the table as the tutorial of Caleb on Livewire Screencast

<img width="1170" alt="CleanShot 2023-09-13 at 14 09 50@2x" src="https://github.com/livewire/livewire/assets/94628273/cc497482-9861-4a81-bc57-5499dba94954">
<img width="1157" alt="CleanShot 2023-09-13 at 14 09 40@2x" src="https://github.com/livewire/livewire/assets/94628273/74283b4b-b3c9-4e8c-823f-3becf5601eed">

**Comment by joffreypersia** [comment]
Up ?

**Comment by PhiloNL** [comment, marked as answer]
Both solutions would work fine. The main difference would be speed, using Alpine will obviously be quicker than requesting the server. So I would test both solutions, maybe throttle network speed as well to get a good sense of what works best when someone has a slower connection
